apps/legacy/src/pages/Swap/components/TransactionDetails.tsx

```diff
@@ -162,6 +229,123 @@ export function TransactionDetails({
+                sx={{
+                  color: 'primary.main',
+                  p: 0.5,
+                  '&.Mui-disabled': { color: 'primary.main' },
+                }}
+                endIcon={
+                  isProviderMenuOpen ? <ChevronUpIcon /> : <ChevronDownIcon />
+                }
+              >
+                <Stack
+                  direction="row"
+                  sx={{
+                    columnGap: 1,
+                    justifyContent: 'flex-end',
+                    alignItems: 'center',
+                  }}
+                >
+                  {manuallySelected == false
+                    ? 'Auto • ' + quote.aggregator.name
+                    : quote.aggregator.name}
+                </Stack>
+              </Button>
+              <Menu
+                anchorEl={selectButtonRef.current}
+                open={isProviderMenuOpen}
+                onClose={() => {
+                  setIsProviderMenuOpen(false);
+                }}
+                PaperProps={{
+                  sx: { width: 220, backgroundColor: 'grey.800', mr: 3 },
+                }}
+                anchorOrigin={{
+                  vertical: 'bottom',
+                  horizontal: 'right',
+                }}
+                transformOrigin={{
+                  vertical: 'top',
+                  horizontal: 'right',
+                }}
+              >
+                {providers
+                  .filter((q) => isMarkrQuote(q.quote))
+                  .map((qq) => {
+                    const q = qq.quote as MarkrQuote;
+                    return (
+                      <MenuItem
+                        key={q.aggregator.id}
+                        data-testid={`provider-${q.aggregator.id}`}
+                        onClick={() => {
+                          setManuallySelected(true);
+                          setQuotes({ ...quotes, selected: qq });
+                          setIsProviderMenuOpen(false);
+                        }}
+                        disableRipple
+                        sx={{ minHeight: 'auto', py: 1 }}
+                      >
+                        <Stack
+                          direction="row"
+                          sx={{
+                            justifyContent: 'space-between',
+                            alignItems: 'center',
+                            width: '100%',
+                          }}
+                        >
+                          <Stack
+                            direction="row"
+                            sx={{
+                              columnGap: 1,
+                              alignItems: 'center',
+                            }}
+                          >
+                            {q.aggregator.id === 'auto' ? (
+                              <AvaCloudConnectIcon size={32} />
+                            ) : (
+                              <img
+                                src={`/images/logos/provider-${q.aggregator.id}.svg`}
```

Comment:
What happens when Markr adds another provider to the orchestrator and we don't have a logo for it?
Should the logos be loaded from Markr's servers instead? Or maybe we need some kind of a fallback icon for them?